### PR TITLE
Add XPath and JSON Pointer path formatters

### LINKED ISSUE #32: Support XPath for XML and JSON Pointer for JSON/YAML in validation output in CLI default and SARIF output modes
# User Story:

As a developer of OSCAL-based tools, in order to transition between different formats more seamlessly in general and processing validation results in particular, I would like the ability to switch from the default Metapaths in constraints' validation results from Metapath to XPath (1.0 or 3.0 flavors TBD) and JSON Pointer

## Goals:

Support for:
- [ ] emitting paths in constraint validation results in pure XPath (1.0 or 3.0 up for discussion and confirmation with oscal-cli maintainers)
- [ ] emitting path in constraint validation results in pure JSON Pointer ([RFC6901](https://datatracker.ietf.org/doc/html/rfc6901))
- [ ] capability to automatically switch between them based on input document type (auto-sense XPath for XML, JSON Pointer for JSON/YAML)
- [ ] a command-line option at runtime to explicitly mandate the output of one over the other

## Dependencies:

{Describe any previous issues or related work that must be completed to start or complete this issue.}

## Acceptance Criteria

- [ ] All website and readme documentation affected by the changes in this issue have been updated.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### PR DESCRIPTION
**Pull request by david-waltermire**
## Summary

This PR adds new path formatters and format-based selection to the Metaschema framework:

- **XPathFormatter** - Produces namespace-qualified XPath 3.1 paths using EQName format
- **JsonPointerFormatter** - Produces RFC 6901 compliant JSON Pointer paths
- **Format-based selection** - Automatically uses appropriate path format based on document type

Addresses metaschema-framework/oscal-cli#32

### Key Features

**Path Formatters:**
- XPath with EQName namespace qualification (`Q{namespace}element[position]`)
- Proper handling of XML grouping (GROUPED/UNGROUPED)
- JSON Pointer with proper handling of all JSON grouping behaviors (KEYED/LIST/SINGLETON_OR_LIST/NONE)
- Support for JSON key flag values in KEYED collections
- RFC 6901 escaping (`~` → `~0`, `/` → `~1`)

**Format-Based Selection:**
- JSON/YAML documents automatically use JSON Pointer paths
- XML documents automatically use XPath paths
- New `--path-format` CLI option to override: `auto`, `metapath`, `xpath`, `jsonpointer`

### Core Module
- Add `XPathFormatter` - XPath 3.1 paths with EQName-qualified names
- Add `JsonPointerFormatter` - RFC 6901 compliant JSON Pointer paths
- Add `PathFormatSelection` enum for path format options
- Add `XPATH_PATH_FORMATTER` and `JSON_POINTER_PATH_FORMATTER` constants to `IPathFormatter`
- Remove obsolete `JsonPathFormatter` placeholder code
- Update package-info.java documentation

### Databind Module
- Add `Format.getPathFormatter()` - returns appropriate formatter for document format
- Add `Format.resolvePathFormatter(PathFormatSelection, Format)` - resolves formatter based on selection

### CLI Module
- Add `--path-format` option to validation commands
- Update `LoggingValidationHandler` to use configurable path formatter

## Test plan

- [x] 26 unit tests for XPathFormatter covering all node types and edge cases
- [x] 21 unit tests for JsonPointerFormatter covering all grouping behaviors and RFC 6901 escaping
- [x] 10 unit tests for Format path formatter resolution
- [x] Full test suite passes
- [x] CI build passes (`mvn install -PCI -Prelease`)

## Summary by CodeRabbit

* **New Features**
  * Added XPath 3.1 EQName-based path formatter for XML paths
  * Added RFC 6901 JSON Pointer formatter for JSON/YAML paths
  * Added `--path-format` CLI option for selecting path output format (auto, metapath, xpath, jsonpointer)
  * Validation output now displays paths in selected format

* **Removals**
  * Removed legacy JsonPathFormatter

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by aj-stein — DISMISSED** [review]
Very useful, I look forward to using this feature.

File: PRDs/20251215-path-formatters/plan.md [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+PathFormatSelection
+    ↓ combined with
+Format (from document or --as option)
+    ↓ resolved via
+IPathFormatter.resolveFormatter()
+    ↓ passed to
+AbstractConstraintValidationHandler.setPathFormatter()
+    ↓ used by
+Validation result messages
+```
```
[the remark this answers is not in the document]
Comment (david-waltermire) [reply PRDs/20251215-path-formatters/plan.md]: done

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/format/JsonPointerFormatter.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  public String formatField(IFieldNodeItem field) {
+    return formatModelItem(field);
+  }
+
+  @Override
+  @NonNull
+  public String formatFlag(IFlagNodeItem flag) {
+    // JSON Pointer does not use @ prefix for attributes
+    return escapeJsonPointer(flag.getQName().getLocalName());
+  }
```
[the remark this answers is not in the document]
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/format/JsonPointerFormatter.java:100]: Override methods inherit docs.
